frontend/src/main/components/Users/UsersTable.js

```diff
@@ -1,33 +1,116 @@
+    const deleteMutation = useBackendMutation(
+        cellToAxiosParamsDelete,
+        { onSuccess: onDeleteSuccess },
+        ["/api/admin/users"]
+    );
+
+    const unhideMutation = useBackendMutation(
+        cellToAxiosParamsUnhide,
+        { onSuccess: onDeleteSuccess },
+        ["/api/admin/users"]
+    );
+    // Stryker restore all
+
+    const deleteCallback = async (cell) => {
+        deleteMutation.mutate(cell);
+    }
+
+    const unhideCallback = async (cell) => {
+        unhideMutation.mutate(cell);
+    }
+
+    const columns = [
+        {
+            Header: 'id',
+            accessor: 'id', // accessor is the "key" in the data
+        },
+        {
+            Header: 'First Name',
+            accessor: 'givenName',
+        },
+        {
+            Header: 'Last Name',
+            accessor: 'familyName',
+        },
+        {
+            Header: 'Email',
+            accessor: 'email',
+        },
+        {
+            Header: 'Admin',
+            id: 'admin',
+            accessor: (row, _rowIndex) => String(row.admin) // hack needed for boolean values to show up
+        },
+        {
+            Header: 'Hidden',
+            id: 'hidden',
+            accessor: (row, _rowIndex) => String(row.hidden) // hack needed for boolean values to show up
+        }
+    ];
+
+    /*
+    const columnsWithHide = [
+        ...columns,
+        ButtonColumn("Hide", "danger", deleteCallback, "UsersTable"),
+        ButtonColumn("Unhide", "success", unhideCallback, "UsersTable")
+    ];
+    */
+
+    // if this row is an admin, disable the hide/unhide buttons
+    // if this row is hidden, disable the hide button
+    // if this row is not hidden, disable the unhide button
+    const columnsWithHide = [
+        ...columns,
+        {
+            Header: 'Hide',
+            id: 'hide',
+            Cell: ({ cell }) => (
+                <Button
+                    variant={"danger"}
+                    onClick={() => deleteCallback(cell)}
+                    data-testid={`UsersTable-cell-row-${cell.row.index}-col-${cell.column.id}-button`}
+                    disabled={cell.row.values["hidden"] === "true" || 
+                    // Stryker disable all : backend does not support admin hiding
+                    cell.row.values["admin"] === "true"
+                    // Stryker restore all
+                    }
+                >
+                    {"Hide"}
+                </Button>
+            )
+        },
+        {
+            Header: 'Unhide',
+            id: 'unhide',
+            Cell: ({ cell }) => (
+                <Button
+                    variant={"success"}
+                    onClick={() => unhideCallback(cell)}
+                    data-testid={`UsersTable-cell-row-${cell.row.index}-col-${cell.column.id}-button`}
+                    disabled={cell.row.values["hidden"] === "false" ||
+                    // Stryker disable all : backend does not support admin hiding
```

Comment:
I don't understand the comment "backend does not support admin hiding".   I mean, I understand the comment, but that's not a reason to exclude this from Stryker mutation testing. In fact, this should most certainly be tested with a frontend tests (one that has an admin user and asserts that the button is disabled, and another than is a non-admin users and asserts that the button is not disabled.)

Reply:
I'm a little confused about this one. The code implies that the backend will always return a user with
1. admin and not hidden
2. user and hidden
3. user and not hidden

So a hidden admin is not considered. Should I just include a test with hidden admin?

```js
    test("admin will always disable both buttons", () => {
        const testId = "UsersTable";
        let baseUser = usersFixtures.threeUsers[0];

        baseUser.admin = true;
        baseUser.hidden = true;
        render(
            <QueryClientProvider client={queryClient}>
                <UsersTable users={[baseUser]}/>
            </QueryClientProvider>
        );

        expect(screen.getByTestId(`${testId}-cell-row-0-col-hide-button`)).toBeDisabled();
        expect(screen.getByTestId(`${testId}-cell-row-0-col-unhide-button`)).toBeDisabled();
      });
```